org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/UiSession.java

```diff
@@ -605,6 +611,52 @@ public void dispose() {
     m_sessionMetrics.sessionDestroyed(SESSION_TYPE);
   }
 
+  protected void runInModelJob(Runnable r, String name) {
+    runInModelJob(r, name, false);
+  }
+
+  protected void runInModelJob(Runnable r, String name, boolean wait) {
+    if (r == null) {
+      return;
+    }
+
+    // current thread is the model thread -> run
+    if (ModelJobs.isModelThread()) {
+      r.run();
+      return;
+    }
+
+    // current thread is NOT the model thread and the client session is active -> run in model job
+    final IClientSession clientSession = getClientSession();
+    if (clientSession != null && clientSession.isActive() && !clientSession.isStopping()) {
+      final ClientRunContext clientRunContext = ClientRunContexts.copyCurrent(true).withSession(clientSession, true);
+      IFuture<Void> modelJob = ModelJobs.schedule(
```

Comment:
More readable?

```
      IFuture<Void> modelJob = ModelJobs.schedule(r::run, ModelJobs.newInput(clientRunContext)
              .withName(name)
              .withExceptionHandling(null, false));// Propagate exception to caller (UIServlet)
```

Reply:
I do not think so. I find the indentation misleading in this case, this suggests that `withName` is a method called on the scheduled job not on the input.
Consider a potential `whenDone` call on a new line at the end. `withName` is called on the result of `ModelJobs.newInput` while `whenDone` is called on the result of `ModelJobs.schedule`, but both lines will have the same indentation.
Therefore, I prefer one level of indentation per block that is not closed on the same line it is opened. In this case one for the `schedule`-method-parameters and one for the `input`-method-chaining.